Thesis.org

```diff
@@ -1,11 +1,725 @@
+Now, in line 11, we define the function that will be called once an action is emitted. Like mentioned earlier, if the action is a ~Commit~, we execute the code in line 13. Here we set our state to the result of the ~continuation~ of the ~Bind~ of our ~inner~ state, by calling ~core/return~ with the ~:state~ keyword. With that we complete our trampoline. Because we set our state, the component will be rendered again, this time with the updated state, which is the next step in our sequential composition, because it is the result of the ~continuation~ of our ~Bind~.
+** show
+To display our sequential composition we can use the ~runner~. If however we just want to display one step (indefinitely), we can use ~show~. This is a simple helper function that is used inside of the ~runner~ to display the ~Bind~. At its core it has a ~cond~ call, which allows us to react to specific conditions. This is necessary, because if we get an ~Item~, we can just display it. If however we get a ~Prog~, we need to unwrap the ~Item~ from it. Furthermore, if we get a ~Bind~, we first need to get the ~Prog~ inside of it, before we can unwrap it. Lastly, to make usage easier, if anything else is passed, we display an empty ~fragment~, which is equivalent to nothing. Getting passed neither ~Prog~ or ~Bind~ is the case after finishing the last ~Bind~. See listing 23 for the entire code of the ~show~ function.
+#+CAPTION: Code for the ~show~ function.
+#+begin_src clojure
+  (defn show
+    [x]
+    {:post [(c/item? %)]}
+    (cond
+      (prog? x) (prog-item x)
+      (bind? x) (prog-item (bind-item x))
+      (c/item? x) x
+      :else (c/fragment)))
+#+end_src
+** Macros
+The available primitives already enable the developer to craft elaborate sequential compositions, however they aren't able to provide the ~let~ like usage on their own. For this reason, we will implement macros for our API. Only a macro version of the ~runner~ is required, but it makes more sense to enhance the ~then~ primitive first and later wrap the ~runner~ with something that takes advantage of the ~then~ macro. Before we dive into the macros themselves, we first need to take a look at how macros work in ClojureScript.
+*** TODO Setup
+Both Clojure and Clojurescript have access to macros, though implementing one is more complex in the later. This is because of the compilation process. Macros are run during the compile process of ClojureScript. In short, this means the compilation process needs to be kept in mind when writing ClojureScript macros. You cannot, for instance, put macros in a ~.cljs~ file.
```

Comment:
In short: Macros are always expanded by Clojure, even though they may produce ClojureScript code.

So a macro on Clojure:
```
ClojureCode -> ClojureCode
```

is run by Clojure. And a macro on ClojureScript:

```
ClojureScript -> ClojureScript
```

is also run by Clojure.

Reply:
```suggestion
Both Clojure and ClojureScript have access to macros, though implementing one is more complex in the later. This is because macros are always expanded by Clojure, even if they may produce ClojureScript code. This means Clojure is always involved, even in pure ClojureScript projects. This means the compilation process needs to be kept in mind when writing ClojureScript macros. You cannot, for instance, put macros in a ~.cljs~ file.
```
I like your "in short" explanation and jammed it in there. I still mention "the compilation process", but I can change that if it's incorrect etc.